### ISSUE TITLE
I encountered a problem while using YOLOv11 for object detection. When I detect a relatively long object, it doesn't get detected fully.

### ISSUE THREAD

**Issue by zwf159** [ultralytics/ultralytics#19523]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am using an input image size of 2560x2560, and the number of training epochs is 700. The model I selected is YOLOv11l, and my optimizer is SGD. During inference, I use SAHI to split the image into 2560-sized patches for inference, with a width and height overlap of 0.7 and 0.2. During prediction, there are some areas at the bottom that are not detected.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zwf159, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) for comprehensive information and usage examples, including detailed [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) guides that may help tackle issues like this.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) that includes your code snippet, a sample image, and detailed steps to reproduce the issue. This will help us debug more efficiently.

If this is a custom training ❓ Question, could you share more information? Specifically:
- Details of your dataset (e.g., sample images or dataset setup)
- Your training command/configuration
- Sample predictions or error outputs
This will help us better understand the issue and provide tailored advice. Also, ensure you've followed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize YOLO's performance.

Join the Ultralytics community where it suits you best! 🎉 For real-time chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧, engage in discussions on [Discourse](https://community.ultralytics.com/), or explore threads on [Subreddit](https://reddit.com/r/Ultralytics) to connect with users tackling similar challenges.

## Upgrade

Ensure you’re running the latest `ultralytics` package to verify the issue isn’t resolved in the latest release. Upgrade your environment by running:

```bash
pip install -U ultralytics
```

## Environments

YOLO is compatible with a range of verified environments. Try any of these for optimized performance:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu environments every 24 hours and upon every commit. ✅

🤖 This is an automated response, but rest assured that an Ultralytics engineer will assist you further shortly!

**Comment by glenn-jocher** [comment]
@zwf159 for elongated object detection challenges with Ultralytics YOLO11, consider these adjustments:

1. **SAHI Configuration**: Increase height overlap ratio (try 0.3-0.4) to better capture vertical objects in adjacent slices. The current 0.2 overlap might be insufficient for objects spanning multiple slices.

2. **Input Resolution**: While using 2560x2560 resolution, ensure your SAHI slice size matches the training input size. YOLO11 performs best when inference conditions match training parameters. See our [SAHI Tiled Inference guide](https://docs.ultralytics.com/guides/sahi-tiled-inference/) for optimization tips.

3. **Model Architecture**: For elongated objects, consider:
   - YOLO-World for open-vocabulary detection capabilities
   - RT-DETR models which handle varied object shapes effectively

4. **Training Parameters**: At 700 epochs, verify you're not overfitting using validation metrics (mAP curves). Consider implementing a cosine learning rate scheduler instead of default SGD settings.

If these adjustments don't resolve the issue, please share a minimal reproducible example via [Ultralytics GitHub Issues](https://github.com/ultralytics/ultralytics/issues) following our [Bug Report Guide](https://docs.ultralytics.com/guides/yolo-common-issues/#minimum-reproducible-example).

**Comment by zwf159** [comment]
> [@zwf159](https://github.com/zwf159) for elongated object detection challenges with Ultralytics YOLO11, consider these adjustments:
> 
> 1. **SAHI Configuration**: Increase height overlap ratio (try 0.3-0.4) to better capture vertical objects in adjacent slices. The current 0.2 overlap might be insufficient for objects spanning multiple slices.
> 2. **Input Resolution**: While using 2560x2560 resolution, ensure your SAHI slice size matches the training input size. YOLO11 performs best when inference conditions match training parameters. See our [SAHI Tiled Inference guide](https://docs.ultralytics.com/guides/sahi-tiled-inference/) for optimization tips.
> 3. **Model Architecture**: For elongated objects, consider:
>    
>    * YOLO-World for open-vocabulary detection capabilities
>    * RT-DETR models which handle varied object shapes effectively
> 4. **Training Parameters**: At 700 epochs, verify you're not overfitting using validation metrics (mAP curves). Consider implementing a cosine learning rate scheduler instead of default SGD settings.
> 
> If these adjustments don't resolve the issue, please share a minimal reproducible example via [Ultralytics GitHub Issues](https://github.com/ultralytics/ultralytics/issues) following our [Bug Report Guide](https://docs.ultralytics.com/guides/yolo-common-issues/#minimum-reproducible-example).

Is this related to regmax?

**Comment by glenn-jocher** [comment]
@lhj5426 for elongated text detection challenges with YOLO11X, we recommend:  
1. Verify potential overfitting by monitoring validation mAP curves during training  
2. Reduce epochs (700 may be excessive for larger models) and implement stronger augmentation  
3. Consider RT-DETR models which show improved performance on aspect ratio extremes  

If issues persist, please create a minimal reproducible example following our Bug Report Guide and share via GitHub Issues.

**Comment by lhj5426** [comment]
> [@_lhj5426_](https://github.com/lhj5426)* for elongated text detection challenges with YOLO11X, we recommend:
> 
> 1. Verify potential overfitting by monitoring validation mAP curves during training
> 2. Reduce epochs (700 may be excessive for larger models) and implement stronger augmentation
> 3. Consider RT-DETR models which show improved performance on aspect ratio extremes
> 
> If issues persist, please create a minimal reproducible example following our Bug Report Guide and share via GitHub Issues.

![Image](https://github.com/user-attachments/assets/dec667f6-a394-4f03-a66c-9506f9760188)
I didn’t train for 700 epochs.  

I only trained for a little over 80 epochs using 27,595 images.  

This is a screenshot from the end of the training.  

Using the YOLO11X model, it just cannot recognize extremely long text.

**Comment by glenn-jocher** [comment]
[ @lhj5426](https://github.com/lhj5426) Thank you for clarifying the epoch count. For elongated text detection with YOLO11X, we recommend:  
1. Verify label consistency for extreme aspect ratio annotations in your dataset  
2. Implement `hsv_h`, `hsv_s`, and `rotate` augmentations specifically for horizontal stretching  
3. Test with [RT-DETR](https://docs.ultralytics.com/models/rtdetr/) models which handle aspect ratio extremes more effectively  

If the issue persists, please share a minimal reproducible example following our [Bug Report Guide](https://docs.ultralytics.com/guides/yolo-common-issues/#minimum-reproducible-example) via GitHub Issues.

**Comment by lhj5426** [comment]
> [ @_lhj5426_](https://github.com/lhj5426)* Thank you for clarifying the epoch count. For elongated text detection with YOLO11X, we recommend:
> 
> 1. Verify label consistency for extreme aspect ratio annotations in your dataset
> 2. Implement `hsv_h`， `hsv_s`，和 `rotate` augmentations specifically for horizontal stretching
> 3. Test with [RT-DETR](https://docs.ultralytics.com/models/rtdetr/)*  models which handle aspect ratio extremes more effectively
> 
> If the issue persists, please share a minimal reproducible example following our [Bug Report Guide](https://docs.ultralytics.com/guides/yolo-common-issues/#minimum-reproducible-example)* via GitHub Issues.
Thank you for your suggestion.
However, the issue is that I am using software developed by someone else for AI-based comic translation.
The software only supports using its own trained YOLO model for OCR pre-processing tasks, such as text object detection before comic translation.
So, I can only use YOLO V8, V11, or V12, or as you mentioned, RT-DETR.
I want to use it, but the software I’m using doesn’t support loading this model and cannot use it

**Comment by glenn-jocher** [comment]
For YOLO11X optimization within third-party software constraints, focus on adjusting anchor ratios in your model configuration to better match extreme aspect ratios of text elements. If the software allows custom YAML configurations, modify anchors to prioritize elongated shapes. For integration limitations, consider reaching out to the software maintainers about adding RT-DETR support via their [GitHub Issues](https://github.com/ultralytics/ultralytics/issues) page.
